### PR TITLE
Table inspector re design

### LINKED ISSUE #1660: Implement new design for Table Inspector
Figma: https://www.figma.com/file/xHb5oIqye3fnXtb2heRH34/Styling?node-id=611%3A3416

### General Items
- [x] Update the designs for the table inspector tabs.
- [x] Update the designs for `Collapsible` components
### Table Mode
- [x] Update the designs for table `properties` section.
- [x] Update the designs for table `links` section.
- [x] Update the designs for table `actions` section.
- [x] Update the designs for table `constraints` section not including the constraints modal. 

### Column Mode
- [x] Update the design of `No Cell selected` view.
- [x] Update the designs for column `properties` section.
- [ ] #1941
- [x] Update the designs for column `default value` section.
- [x] Update the designs for column `actions` section.

### Row Mode
- [x] Update the design of `No Row selected` view.
- [x] Update the designs for row `actions` section.

### LINKED ISSUE #1960: Pick better columns to show within the list of table links
## Reproduce

1. Use the Library Management schema.

1. Open the Publication table page and look at the Links section within the table inspector.

    image

1. Expect the column shown under "Authors" to be "Author", but observe it to be "id"
1. Expect the column shown under "Publishers" to be "Publisher", but observe it to be "id"

CC @rajatvijay

### PR DESCRIPTION
**Pull request by rajatvijay**
Fixes #1660
Fixes #1960

What this PR includes
Implements the new designs for the schema page as mentioned here: https://www.figma.com/file/xHb5oIqye3fnXtb2heRH34/Styling?node-id=769%3A10804 except for https://github.com/centerofci/mathesar/issues/1941

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by pavish** [comment]
I'm assigning myself as the reviewer of this PR since it deals with type change logic, and I'm also assigned as the helper in the spreadsheet.

File: mathesar_ui/src/stores/table-data/records.ts
Comment (seancolsen) [inline mathesar_ui/src/stores/table-data/records.ts:693]: Why is this pure function defined as a method on a class to which it does not directly relate? I think this file is a good place to put it, but I would expect it to be defined outside the class.

I thought we actually had a linting rule agains this. Not sure what happened to that.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/LinkSection.svelte
Comment (seancolsen) [inline mathesar_ui/src/systems/table-view/table-inspector/table/links/LinkSection.svelte:19]: It would be nice to convert these IIFEs to ES6 syntax

File: mathesar_ui/src/systems/table-view/table-inspector/table/TableActions.svelte
Comment (seancolsen) [inline mathesar_ui/src/systems/table-view/table-inspector/table/TableActions.svelte:50]: This is very minor, but here's a case where I don't like Prettier's auto-formatting choices. Prettier is trying to ensure that the code does not have any white space between the opening tag and the first word, or between the last word and the closing tag. That sort of thing _can_ matter, but only in very rare cases. So when I end up with code that Prettier breaks up in this way, I change it to the following, which I find more readable:

```html
<Help>
  Open this table in Data Explorer to query and analyze your data.
</Help>
```

File: mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte
Comment (seancolsen) [inline mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte:6]: Instead of `isDBLevelConfiguration`, we should be using `isDbLevelConfiguration`, per our code standards to treat acronyms as words. Note that we still have some older code (e.g. `postAPI`) which we wrote before adopting that code standard, but it would be nice to try adhering to this standard going forward.

File: mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte
Comment (seancolsen) [inline mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte:25]:
```suggestion
<slot slot="content">
  This section affects the configuration of the database.
</slot>
```

File: mathesar_ui/src/systems/table-view/table-inspector/TableInspector.svelte
Comment (seancolsen) [inline mathesar_ui/src/systems/table-view/table-inspector/TableInspector.svelte:74]: All this global CSS makes me suspicious that the `<TabContainer>` component may not be designed well, though I haven't looked at it deeply enough to figure out what the issue is here. But this is to say, I wouldn't expect to see all this global CSS for a seemingly straightforward use-case of the tabs system. Feel free to disregard this comment if you've already given this some thought.

**Review by seancolsen — COMMENTED** [review]
I looked over the diff and wanted to mention a few things alongside whatever review @pavish gives.

File: mathesar_ui/src/stores/table-data/records.ts
Comment (seancolsen) [reply mathesar_ui/src/stores/table-data/records.ts:693]: I found the eslint rule: [class-methods-use-this](https://eslint.org/docs/latest/rules/class-methods-use-this)

It looks like @pavish disabled it in #1721. Any comments about why, Pavish? I do kind of appreciate that rule.

File: mathesar_ui/src/component-library/help/Help.scss
Comment (seancolsen) [inline mathesar_ui/src/component-library/help/Help.scss:6]: Also, this change turns help bubbles red, making them look like errors or warnings. Blue is a color that seems to be commonly associated with help messages. Can we keep it blue (and find a suitable on-brand shade of blue)?

**Comment by seancolsen** [comment]
I just wanted to note that I've rebased #1901 so that I'm working on top of this branch now. I may find some other things to mention about this PR as I continue to work with it, but no need to hold up merging for any further review from me. All the things I mentioned in my above review can be handled post-merge.

File: mathesar_ui/src/component-library/help/Help.scss
Comment (kgodey) [reply mathesar_ui/src/component-library/help/Help.scss:6]: Help bubbles are red in the Figma design, this should be an action item for @ghislaineguerin.

**Comment by seancolsen** [comment]
I opened #1959 for another issue I found.

File: mathesar_ui/src/systems/table-view/table-inspector/ActionItem.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/ActionItem.svelte:75]: I don't think that we need this component & the abstraction it provides. We already have common components that perform the operations that are needed here, and they contain the required styling for states such as hover/focus/active etc., which this component lacks.

Use the `Button` component directly for all action items which needs to be buttons, and use the `AnchorButton` component for all anchors. Update AnchorButton so that it uses the same styles/appearances from Button.

File: mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte:19]: Why is the help button a dropdown?

As per the design, it should be a tooltip, which appears on hover. @ghislaineguerin Am I correct in assuming this?

This help content currently only appears when clicking on the database icon.

File: mathesar_ui/src/systems/table-view/table-inspector/column/ColumnOptions.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/column/ColumnOptions.svelte:93]: Why is a Checkbox enclosed within button here (and in a few other places)?

Use the `LabelledInput` component if the purpose here is to show the label inline.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:47]: This logic is incorrect. We should not determine link direction based on the `multiple_results` property. It needs to be determined based on the second array element of `fk_path` property being `true`.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:26]: Identification logic for column is incorrect. This logic works only for links from `from_other_tables`.

For links in the base table i.e `in_this_table`, the first element in the array represents the column id, not the second. This check is direction and `fk_path[0][2]` needs to be taken into account to determine which column id to pick.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/LinkSection.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/table/links/LinkSection.svelte:56]: The figma design has this button in the content with a different styling, and it is not as part of the header.

File: mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte:19]: The input to set description is missing here.

File: mathesar_ui/src/systems/table-view/table-inspector/table/AdvancedActions.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/table/AdvancedActions.svelte:21]: Please create an issue to keep track of this.

**Comment by seancolsen** [comment]
I opened #1960 for another issue I found.

File: mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte:41]: Open the Records tab, and try creating a new record. The uncaught error breaks the entire product.

This is an unsafe access, where saved records may not have the row index when new records are present. There is a method called `getRecordRows` in records.ts, which should be used for any record access. Undefined return should be assumed on any index access on arrays, where length check isn't done.

**Review by pavish — CHANGES_REQUESTED** [review]

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (pavish) [reply mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:26]: I see that @seancolsen has opened an issue which is caused by this incorrect logic: https://github.com/centerofci/mathesar/issues/1960.

I'd like the fix to be made on this PR itself. @rajatvijay Please mark the issue as fixed/closed once you've implemented the fix here.

File: mathesar_ui/src/component-library/help/Help.scss
Comment (ghislaineguerin) [reply mathesar_ui/src/component-library/help/Help.scss:6]: I'd prefer that we use any of the gray (slate) shadings instead of blue. In most examples I've found of other apps it's always a color close to that of the body text.

File: mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte
Comment (seancolsen) [reply mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte:41]: > Undefined return should be assumed on any index access on arrays, where length check isn't done.

This has been bugging me for a while, and I've opened #1966 in search of a more systemic solution.

File: mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte
Comment (pavish) [reply mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte:19]: We do support table descriptions. It's implemented on the backend.

File: mathesar_ui/src/stores/table-data/records.ts
Comment (seancolsen) [reply mathesar_ui/src/stores/table-data/records.ts]: Looks good. Thanks.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/LinkSection.svelte
Comment (seancolsen) [reply mathesar_ui/src/systems/table-view/table-inspector/table/links/LinkSection.svelte]: Looks good. Thanks.

File: mathesar_ui/src/systems/table-view/table-inspector/table/TableActions.svelte
Comment (seancolsen) [reply mathesar_ui/src/systems/table-view/table-inspector/table/TableActions.svelte]: Looks good. Thanks.

File: mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte
Comment (seancolsen) [reply mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte]: Looks good. Thanks.

File: mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte
Comment (seancolsen) [reply mathesar_ui/src/systems/table-view/table-inspector/CollapsibleHeader.svelte]: Looks good. Thanks.

File: mathesar_ui/src/systems/table-view/table-inspector/TableInspector.svelte
Comment (seancolsen) [reply mathesar_ui/src/systems/table-view/table-inspector/TableInspector.svelte]: I see there is still some global CSS in there now, but this is much better than before.

File: mathesar_ui/src/component-library/help/Help.scss
Comment (seancolsen) [reply mathesar_ui/src/component-library/help/Help.scss]: Looks good. Thanks.

**Comment by kgodey** [comment]
If this is waiting on @pavish's review, please assign to @pavish and update status label to `review`.

**Comment by rajatvijay** [comment]
@kgodey there was a small thing that I waiting for @pavish's help with. It is resolved now, hence assigned the review now.

File: mathesar_ui/src/component-library/icon/Icon.svelte
Comment (pavish) [inline mathesar_ui/src/component-library/icon/Icon.svelte:76]: I'm okay with using the title property as standby icon tooltips, but the actual usecase requires us to have a tooltip component, which isn't reserved to icons.

Please create a separate issue for this.

File: mathesar_ui/src/pages/table/TablePage.svelte
Comment (pavish) [inline mathesar_ui/src/pages/table/TablePage.svelte:15]: This component is no longer present. Please merge changes from the latest master and resolve conflicts.

File: mathesar_ui/src/stores/tables.ts
Comment (pavish) [inline mathesar_ui/src/stores/tables.ts:423]: `max_depth` should be configurable and must be accepted as a function parameter, instead of being hardcoded as 1.

File: mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte:8]: @rajatvijay Please ensure that all imports of the component library from mathesar code happens through `@mathesar-component-library` alias. Automatic imports would prefer directly accessing the directory directly and we do not have a way to force the alias as of the moment, expect manually.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:21]: Please remove this console statement

**Review by pavish — CHANGES_REQUESTED** [review]

File: mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte
Comment (pavish) [reply mathesar_ui/src/systems/table-view/table-inspector/record/RowActions.svelte:8]: No change needed in this PR, we can correct the existing imports later. However, please ensure to follow this in upcoming PRs.

File: mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte
Comment (pavish) [reply mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte:19]: @rajatvijay Please create a separate issue for this if you're planning to handle this in a different PR.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (pavish) [inline mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:29]: Always assume that these properties could be undefined.

i.e. `joinableTablesResult.tables[linkedTable.target]` and `joinableTablesResult.columns[columnId]`.

The API currently does not provide the column & table details of the base table that is part of the request. As a result this will always result in an error when there are FK links to other tables in the base table. Was this code tested thoroughly?

**Review by pavish — CHANGES_REQUESTED** [review]

**Comment by pavish** [comment]
@rajatvijay All my previous review comments have been resolved, except the [table description one](https://github.com/centerofci/mathesar/pull/1876#discussion_r1027408826), feel free to create an issue for it to handle later.

However, I have added a few comments on the newer commits, I'd like them resolved. It also seems that there are some failing tests & typecheck errors due to recent merge with master where LayoutWithHeader2 is no longer present.

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (rajatvijay) [reply mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:29]: >The API currently does not provide the column & table details of the base table that is part of the request. As a result this will always result in an error when there are FK links to other tables in the base table. Please test the code and check API responses after implementing the logic.

@pavish do we have any documentation for this or any other APIs for that matter?

File: mathesar_ui/src/component-library/icon/Icon.svelte
Comment (rajatvijay) [reply mathesar_ui/src/component-library/icon/Icon.svelte:76]: https://github.com/centerofci/mathesar/issues/1982

File: mathesar_ui/src/pages/table/TablePage.svelte
Comment (rajatvijay) [reply mathesar_ui/src/pages/table/TablePage.svelte:15]: d80e96b1

File: mathesar_ui/src/stores/tables.ts
Comment (rajatvijay) [reply mathesar_ui/src/stores/tables.ts:423]: c4f7eb96

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (rajatvijay) [reply mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:21]: 371134d7

File: mathesar_ui/src/pages/table/TablePage.svelte
Comment (rajatvijay) [reply mathesar_ui/src/pages/table/TablePage.svelte]: d80e96b1

File: mathesar_ui/src/stores/tables.ts
Comment (rajatvijay) [reply mathesar_ui/src/stores/tables.ts]: c4f7eb96

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (rajatvijay) [reply mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts]: 371134d7

File: mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts
Comment (pavish) [reply mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts:28]: We do not have any comprehensive documentation. We use the browsable API view provided by Django to get the request/response structure which is often sufficient. If you need help with understanding a API better, it's quicker to get in touch with the backend end.

Comment (rajatvijay) [reply mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts]: 9e47548c

File: mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte
Comment (pavish) [reply mathesar_ui/src/systems/table-view/table-inspector/table/TableMode.svelte:19]: I created https://github.com/centerofci/mathesar/issues/1990 for this

**Review by pavish — APPROVED** [review]
@rajatvijay Looks great!

I added a small commit https://github.com/centerofci/mathesar/commit/9e47548c300346a38a0355c4215308f10e8a91fe to remove additional gap between collapsibles in the inspector's table mode.

**Pull request merged**
